thesis/03-implementation.tex

```diff
@@ -171,6 +171,352 @@ \subsection{Components overview}
 \subsection{Algorithms}
 % Ciekawsze algorytmy, aspekty, mechanizmy np. logowanie, indeksowane, cachowanie, synchronizacja, backup, jakieś procesy w tle, jakieś progress bary, jakieś analizy, generowanie warstw GISowych, lokalizacja etc. Jest tu często o czym pisać. %
 
+\subsubsection{Solver}
+
+\textit{Please note that most of the ideas in this section come from \cite{origami-simulator:paper}}.
+\smallskip
+
+The input to the Solver is a \textit{.fold} file containing an Instruction, and its output is a \textit{.fold} file containing a Guide. 
+Instruction, at its simplest, is a set of vertices laying on a flat surface, and a set of edge assignments in each step.
+The Solver's job is to output a resulting Guide file which has all the information necessary to display
+a folding animation - that is, all the vertex positions at each step in time.
+
+\smallskip
+\textbf{Triangulation}
+\smallskip
+
+The Solver works using only triangular faces.
+Therefore, before any folding-algorithm can happen, it has to generate a mesh that only consists of triangles.
+\begin{figure}[H]
+    \centering
+    \subfloat[\centering Crease pattern before triangulation]{{\includegraphics[width=.4\linewidth]{assets/3-2d-cp.png} }}
+    \qquad
+    \subfloat[\centering Crease pattern after triangulation]{{\includegraphics[width=.4\linewidth]{assets/3-2d-cp-triangulated.png} }}
+\end{figure}
+
+As an input, we get a set of 3D points, but triangulation works only in 2D.
+Because of that first we convert a set of 3D points to 2D by removing one of the axes (the one which will not lead to degenerate faces, if possible).
```

Comment:
> Because of that first we convert a

Sounds a bit weird, and definitely lacks a comma. Frankly, I have no idea whether it is "Because of that, first we" or "Because of that first, we".

Maybe:

> Hence, at first we have to convert

> Therefore, at first we have to convert